pyrsistent/_field_common.py

```diff
@@ -91,11 +88,15 @@ def field(type=PFIELD_NO_TYPE, invariant=PFIELD_NO_INVARIANT, initial=PFIELD_NO_
     :param serializer: function that returns a serialized version of the field
     """
 
-    if isinstance(type, Iterable) and not isinstance(type, six.string_types) and not isenum(type):
-        # Enums and strings are iterable types
-        types = set(type)
+    # NB: We have to check this predicate separately from the predicates in
+    # `maybe_type_to_list` et al. because this one is related to supporting the
+    # argspec for `field`, while those are related to supporting the valid ways
+    # to specify types.
+    # Multiple types must be passed in a tuple or a list.
+    if isinstance(type, (tuple, list)):
```

Comment:
this could break legacy users? really not sure. We shouldn't check against `preserved_iterable` here because then that's mixing the semantics for the argspec and the actual type specification too much, in my opinion, so the right thing to do is to nail down that only tuples and lists are acceptable here.

This could probably lead to weird error messages, since we have to just have a fallthrough `else` clause, but we might be able to solve that by raising/validating inside `maybe_type_to_list` and then re-raising informatively here. And the status quo was leading to weird error messages anyway.

Reply:
I should also add `set` to the list of permissible containers...